src/Data/Argonaut/Decode/Generic/Rep.purs

```diff
@@ -87,3 +91,32 @@ genericDecodeJson = map Rep.to <<< decodeRep
 
 mFail :: forall a. String -> Maybe a -> Either String a
 mFail msg = maybe (Left msg) Right
+
+-- | A function for decoding `Generic` sum types using string literal representations
+-- | Takes a function for transforming the tag name in encoding
+decodeLiteralSum :: forall a r. Rep.Generic a r => DecodeLiteral r => (String -> String) -> Json -> Either String a
+decodeLiteralSum tagNameTransform = map Rep.to <<< decodeLiteral tagNameTransform
+
+class DecodeLiteral r where
+  decodeLiteral :: (String -> String) -> Json -> Either String r
+
+instance decodeLiteralSumInst :: (DecodeLiteral a, DecodeLiteral b) => DecodeLiteral (Rep.Sum a b) where
+  decodeLiteral tagNameTransform j = Rep.Inl <$> decodeLiteral tagNameTransform j <|> Rep.Inr <$> decodeLiteral tagNameTransform j
+
+instance decodeLiteralConstructor :: (IsSymbol name) => DecodeLiteral (Rep.Constructor name (Rep.NoArguments)) where
+  decodeLiteral tagNameTransform j = do
+    let name = reflectSymbol (SProxy :: SProxy name)
+    let decodingErr msg = "When decoding a " <> name <> ": " <> msg
+    tag <- mFail (decodingErr "could not read string for constructor") (toString j)
+    when (tag /= tagNameTransform name) $
+      Left $ decodingErr "string literal " <> tag <> " had an incorrect value."
+    pure $ Rep.Constructor (Rep.NoArguments)
+
+type FailMessage = """
+  `decodeLiteralSum` can't be used without sum types, where all of the constructors aren't unary. This is because a string literal cannot be encoded into a product type.
```

Comment:
Hello multi-negative! 😜 

How about something like:

```
type FailMessage = """`decodeLiteralSum` can only be used with sum types where all of the constructors are unary. A string literal cannot sensibly encode a product type."""
```

for these messages?